File: server/src/services/logging.ts
import { prisma } from "../db.js";

export async function logOperation(input: {
  userId?: number | null;
  stegoFileId?: number | null;
  operationType: string;
  operationStatus: string;
  message: string;
}) {
  return prisma.operationLog.create({
    data: {
      userId: input.userId ?? null,
      stegoFileId: input.stegoFileId ?? null,
      operationType: input.operationType,
      operationStatus: input.operationStatus,
      message: input.message
    }
  });
}

File: app/services/logging.py
from __future__ import annotations

from sqlalchemy.orm import Session

from app.models import OperationLog


def log_operation(
    db: Session,
    *,
    user_id: int | None,
    operation_type: str,
    operation_status: str,
    message: str,
    stego_file_id: int | None = None,
) -> OperationLog:
    log = OperationLog(
        user_id=user_id,
        stego_file_id=stego_file_id,
        operation_type=operation_type,
        operation_status=operation_status,
        message=message,
    )
    db.add(log)
    return log

